src/haddock/modules/topology/topoaa/__init__.py

```diff
@@ -12,55 +11,67 @@
 from haddock.libs.libontology import Format, ModuleIO, PDBFile, TopologyFile
 from haddock.libs.libstructure import make_molecules
 from haddock.libs.libsubprocess import CNSJob
-from haddock.modules import BaseHaddockModule, get_engine
+from haddock.modules import get_engine
+from haddock.modules.base_cns_module import BaseCNSModule
 
 
 RECIPE_PATH = Path(__file__).resolve().parent
 DEFAULT_CONFIG = Path(RECIPE_PATH, "defaults.cfg")
 
 
-def generate_topology(input_pdb, step_path, recipe_str, defaults, mol_params,
-                      protonation=None):
+def generate_topology(
```

Comment:
Shouldn't this be handled by `libcns`?

Reply:
I will see. As we discuss on slack, I need to look to some of these things better. Actually, if `generate_topology` refers only to the `topoaa`, it should stay in `topoaa` until required by any other module maybe. Or instead directly go to `libcns` as you said. That will be other PR.